File: dsa/DSA03006.cs
using System;
using System.Linq;
using System.Collections.Generic;

class Solution {

	static bool CanGreedySorting(List<int> array) {

		List<int> sorted = array.OrderBy(x => x).ToList();

		int left = 0, right = array.Count - 1;
		while (left < right) {

			bool condition1 =
			    array[left] == sorted[left] && array[right] == sorted[right];

			bool condition2 =
			    array[left] == sorted[right] && array[right] == sorted[left];

			if (!(condition1 || condition2)) {
				return false;
			}

			left++;
			right--;
		}

		return true;
	}

	static void Main(string[] args) {

		uint cases;
		uint.TryParse(Console.ReadLine(), out cases);

		while (cases-- > 0) {

			int size = int.Parse(Console.ReadLine());

			List<int> array = Console.ReadLine()
			                      .Split()
			                      .Take(size)
			                      .Select(int.Parse)
			                      .ToList();

			bool result = CanGreedySorting(array);

			Console.WriteLine(result ? "Yes" : "No");
		}
	}
}

File: dsa/DSA03006.py
from typing import List


def canGreedySorting(array: List[int]) -> bool:

    sortedArray = sorted(array)
    left, right = 0, len(array) - 1

    while left < right:
        arrayPair = {array[left], array[right]}
        sortedPair = {sortedArray[left], sortedArray[right]}

        if arrayPair != sortedPair:
            return False

        left, right = left + 1, right - 1

    return True


cases = int(input())

for _ in range(cases):
    size__ = int(input())
    array = list(map(int, input().split()))

    print("Yes" if canGreedySorting(array) else "No")
